Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift

```diff
@@ -744,6 +744,49 @@ public class ParseDiagnosticsGenerator: SyntaxAnyVisitor {
     if shouldSkip(node) {
       return .skipChildren
     }
+    if let unexpected = node.unexpectedBetweenReturnClauseAndInKeyword,
+      let tokens = unexpected.onlyPresentTokens(satisfying: { _ in true }),
+      !tokens.isEmpty
+    {
+      addDiagnostic(
+        unexpected,
+        .unexpectedTokensPriorToIn,
+        handledNodes: [unexpected.id]
+      )
+      let baseInKeyword: TokenSyntax =
+        node.inKeyword.isMissing
+        ? node.inKeyword
+        : node.inKeyword.with(\.presence, .missing)
+      let adjustedInKeyword =
+        baseInKeyword
+        .with(\.leadingTrivia, tokens.first?.leadingTrivia ?? baseInKeyword.leadingTrivia)
+        .with(\.trailingTrivia, [])
+      addDiagnostic(
+        unexpected,
+        .expectedInAfterClosureSignature,
+        fixIts: [
+          FixIt(
+            message: InsertTokenFixIt(missingNodes: [Syntax(adjustedInKeyword)]),
+            changes: [.makePresent(adjustedInKeyword)]
+          )
+        ],
+        handledNodes: [adjustedInKeyword.id]
+      )
+    }
+
+    if node.inKeyword.isMissing {
+      addDiagnostic(
+        node.inKeyword,
+        .expectedInAfterClosureSignature,
```

Comment:
I thought missing tokens are also automatically handled (`ParseDiagnosticsGenerator.handleMissingToken(_:)`). What would happen without this change?